### ISSUE TITLE
Set Prometheus Resource limits from cluster enable page

### ISSUE THREAD

**Issue by cloudnautique** [rancher/rancher#18693]
**What kind of request is this (question/bug/enhancement/feature request):**
Enhancement

**Steps to reproduce (least amount of steps as possible):**
The memory resource limit on the Prometheus instance is low by default. In a cluster with 50 nodes, you need close 1.5GB of Ram on the Prometheus pod to collect cluster metrics. The pod starts running into OOM killed issues very quickly.

The resource limits should be configurable from the page where the user enables monitoring.

**Environment information**
rancher/rancher:v2.2.0-rc3

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): RKE Custom
- Machine type (cloud/VM/metal) and specifications (CPU/memory): Google Compute Engine n1-standard-1

50 nodes.
k8s-1.13

**Comment by cloudnautique** [comment]
Also, to reinforce the settings being configurable here, any changes on the deployment are overwritten when make a change at the monitoring enable page. For instance changing the retention period, resets the updated resource limit settings on the container.

**Comment by laoshancun** [comment]
SGTM.
Now the resource limit will be reset to 500Mi when we add `prometheus.resources.core.limits.memory=2048Mi` to catalog app `cluster-monitoring`.

**Comment by davidnuzik** [comment]
Version: Master 2.2 (3/8/19) bda75638b43f
Used RKE on EC2, k8s v1.13.4
10 worker nodes, test scaling up (to 45 worker nodes) upon increasing prometheus memory. (Ran into machine limit for ec2)
Confirmed enhancement.

Validate default and custom settings upon form submit:
![Screen Shot 2019-03-08 at 3 14 54 PM](https://user-images.githubusercontent.com/45179589/54059108-3a080180-41b5-11e9-9f7e-e5fb99a2e5da.png)
```
prometheus.resources.core.limits.cpu: "1111m"
prometheus.resources.core.limits.memory: "499Mi"
prometheus.resources.core.requests.cpu: "222m"
prometheus.resources.core.requests.memory: "150Mi"
```
Validate settings made it in API.

Validate settings in prometheus-cluster-monitoring Security & Host Config in UI:
![image](https://user-images.githubusercontent.com/45179589/54059350-0c6f8800-41b6-11e9-9e1a-4a7b92a14f4f.png)
(Also check prometheus container)

Validate that when we edit a setting for monitoring config form resends all fields, all values get updated. Ensure http/API/UI look correct.
Increased limits took effect:
![image](https://user-images.githubusercontent.com/45179589/54059691-468d5980-41b7-11e9-88b9-241d9f41ea0f.png)

Scale up nodes significantly to about 45 (ran into machine limit for ec2). Ensure monitoring is still working as expected (no OOM issues, etc); smoke test monitoring.
I observed memory utilization mainly on prometheus container and prometheus-cluster-monitoring pod to get an idea on memory utilization, etc. I smoke tested and did not observe any issues.
